truewiki/storage/gitlab.py

```diff
@@ -0,0 +1,98 @@
+import base64
+import click
+import logging
+import tempfile
+import urllib
+
+from openttd_helpers import click_helper
+
+from .git import Storage as GitStorage
+from .github import OutOfProcessStorage as GitHubOutOfProcessStorage
+
+log = logging.getLogger(__name__)
+
+_gitlab_private_key = None
+_gitlab_url = None
+_gitlab_history_url = None
+
+
+class OutOfProcessStorage(GitHubOutOfProcessStorage):
+    name = "GitLab"
+
+
+class Storage(GitStorage):
+    out_of_process_class = OutOfProcessStorage
+
+    def __init__(self):
+        super().__init__()
+
+        # We need to write the private key to disk: GitPython can only use
+        # SSH-keys that are written on disk.
+        if _gitlab_private_key:
+            self._gitlab_private_key_file = tempfile.NamedTemporaryFile()
```

Comment:
Pedantic: file with private key information is written, but I don't easily see where it is cleaned up. Perhaps add some documentation to make this more clear?

Reply:
`If delete is true (the default), the file is deleted as soon as it is closed`
https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Part of Python documentation already :D